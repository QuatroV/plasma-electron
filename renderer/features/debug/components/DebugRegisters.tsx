import { useState } from "react";

import clsxm from "../../../utils/clsxm";
import useDebugStore from "../stores/debugStore";

const REGISTERS = [
  "EAX",
  "EBX",
  "ECX",
  "EDX",
  "ESI",
  "EDI",
  "ESP",
  "EBP",
  "CS",
  "SS",
  "DS",
  "ES",
  "FS",
  "GS",
  "EIP",
];

const DebugRegisters = () => {
  const registers = useDebugStore((state) => state.registers);

  return (
    <div>
      <div className="px-2 py-1 text-sm font-bold">Registers</div>
      <div className="px-2">
        <table className="w-full table-fixed  text-xs">
          <thead className="  text-xs uppercase text-gray-700 shadow dark:bg-gray-700 dark:text-gray-400">
            <tr>
              <th className=" rounded-tl bg-gray-50 py-1 pl-4 text-start">
                Register
              </th>
              <th className="  flex-row items-center justify-between rounded-tr bg-gray-50 py-1 pl-4 text-start">
                <div>Hex</div>
              </th>
              <th className="  flex-row items-center justify-between rounded-tr bg-gray-50 py-1 pl-4 text-start">
                <div>Info</div>
              </th>
            </tr>
          </thead>
          <tbody className="border border-gray-300">
            {REGISTERS.map((register, idx) => {
              const registerValue = registers[register.toLowerCase()];
              return (
                <tr
                  className={clsxm(
                    idx % 2 === 0 ? "bg-gray-200" : "bg-gray-100",
                    "cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300",
                  )}
                >
                  <td className="pl-4">{register}</td>
                  <td className="pl-4">{registerValue?.hexValue}</td>
                  <td className="pl-4">{registerValue?.info}</td>
                </tr>
              );
            })}
          </tbody>
        </table>
      </div>
    </div>
  );
};

export default DebugRegisters;
