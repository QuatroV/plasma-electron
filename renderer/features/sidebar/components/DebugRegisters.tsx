import { useState } from "react";

const DebugRegisters = () => {
  const [valueForm, setValueForm] = useState<"bin" | "hex">("hex");
  return (
    <div>
      <div className="px-2 py-1 text-sm font-bold">Registers</div>
      <div className="px-2">
        <table className="w-full table-fixed  text-xs">
          <thead className="  text-xs uppercase text-gray-700 shadow dark:bg-gray-700 dark:text-gray-400">
            <tr>
              <th className=" rounded-tl bg-gray-50 py-1 pl-4 text-start">
                Register
              </th>
              <th className=" flex flex-row items-center justify-between rounded-tr bg-gray-50 py-1 pl-4 text-start">
                <div>Value</div>
                <div
                  className=" mr-1 cursor-pointer rounded border border-gray-500 px-1 active:bg-gray-500 active:text-white"
                  onClick={() =>
                    setValueForm((prevState) =>
                      prevState === "hex" ? "bin" : "hex"
                    )
                  }
                >
                  {valueForm === "hex" ? "hex" : "bin"}
                </div>
              </th>
            </tr>
          </thead>
          <tbody className="border border-gray-300">
            <tr className="cursor-pointer border-b border-solid border-gray-300 bg-gray-100 transition-all hover:bg-gray-300">
              <td className="pl-4">EAX</td>
              <td className="pl-4">0x11</td>
            </tr>
            <tr className="cursor-pointer  border-b border-solid border-gray-300 bg-gray-200 transition-all hover:bg-gray-300">
              <td className=" pl-4">EBX</td>
              <td className=" pl-4">0x4F</td>
            </tr>
            <tr className="cursor-pointer  border-b border-solid border-gray-300 bg-gray-100 transition-all hover:bg-gray-300">
              <td className=" pl-4">ECX</td>
              <td className=" pl-4">0xFF</td>
            </tr>
            <tr className="cursor-pointer border-b border-solid border-gray-300  bg-gray-200 transition-all hover:bg-gray-300">
              <td className=" pl-4">EDX</td>
              <td className=" pl-4">0x4F</td>
            </tr>
            <tr className="cursor-pointer border-b border-solid border-gray-300 bg-gray-100 transition-all hover:bg-gray-300">
              <td className=" pl-4">ESP</td>
              <td className=" pl-4">0x4F</td>
            </tr>
            <tr className="cursor-pointer border-b border-solid border-gray-300 bg-gray-200 transition-all hover:bg-gray-300">
              <td className=" pl-4">EBP</td>
              <td className=" pl-4">0x4F</td>
            </tr>
            <tr className="cursor-pointer border-b border-solid border-gray-300 bg-gray-100 transition-all hover:bg-gray-300">
              <td className=" pl-4">ESI</td>
              <td className=" pl-4">0x4F</td>
            </tr>
            <tr className="cursor-pointer border-b border-solid border-gray-300 bg-gray-200 transition-all hover:bg-gray-300">
              <td className=" pl-4">EDI</td>
              <td className=" pl-4">0x4F</td>
            </tr>
            <tr className="cursor-pointer border-b border-solid border-gray-300 bg-gray-100 transition-all hover:bg-gray-300">
              <td className=" pl-4">EFLAGS</td>
              <td className=" pl-4">0x4F</td>
            </tr>
            <tr className="cursor-pointer border-b border-solid border-gray-300 bg-gray-200 transition-all hover:bg-gray-300">
              <td className=" pl-4">CS</td>
              <td className=" pl-4">0x4F</td>
            </tr>
            <tr className="cursor-pointer border-b border-solid border-gray-300 bg-gray-100 transition-all hover:bg-gray-300">
              <td className=" pl-4">SS</td>
              <td className=" pl-4">0x4F</td>
            </tr>
            <tr className="cursor-pointer border-b border-solid border-gray-300 bg-gray-200 transition-all hover:bg-gray-300">
              <td className=" pl-4">DS</td>
              <td className=" pl-4">0x4F</td>
            </tr>
            <tr className="cursor-pointer border-b border-solid border-gray-300 bg-gray-100 transition-all hover:bg-gray-300">
              <td className=" pl-4">ES</td>
              <td className=" pl-4">0x4F</td>
            </tr>
            <tr className="cursor-pointer border-b border-solid border-gray-300 bg-gray-200 transition-all hover:bg-gray-300">
              <td className=" pl-4">FS</td>
              <td className=" pl-4">0x4F</td>
            </tr>
            <tr className="cursor-pointer rounded-b border-solid border-gray-300 bg-gray-100 transition-all hover:bg-gray-300">
              <td className=" pl-4">GS</td>
              <td className=" pl-4">0x4F</td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  );
};

export default DebugRegisters;
