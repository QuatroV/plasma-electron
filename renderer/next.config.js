module.exports = {
  webpack: (config, { isServer }) => {
    if (!isServer) {
      config.target = 'electron-renderer';
    }
    
    config.module.rules.push({
      test: /\.svg$/i,
      issuer: /\.[jt]sx?$/,
      use: ['@svgr/webpack'],
    })

    return config;
  },
};
