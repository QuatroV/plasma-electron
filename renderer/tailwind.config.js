/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    './renderer/pages/**/*.{js,ts,jsx,tsx}',
    './renderer/components/**/*.{js,ts,jsx,tsx}',
    './renderer/features/**/*.{js,ts,jsx,tsx}',
  ],
  theme: {
    extend: {
      backgroundImage: {
        'gradient-radial': 'radial-gradient(var(--tw-gradient-stops))',
      },
      keyframes: {
        "slow-appear": {
          '0%': { transform: 'scale(0.1, 1)', transform: 'translate(-5%, 0)', opacity: "0.1" },
          '100%': { transform: 'scale(1, 1)', transform: 'translate(0, 0)', opacity: "1" },
        },
        "slow-appear-no-opacity": {
          '0%': { transform: 'scale(0.1, 1)', transform: 'translate(-5%, 0)'},
          '100%': { transform: 'scale(1, 1)', transform: 'translate(0, 0)'},
        },
        "roll": {
          from: { transform: 'rotate(0deg)'},
          to: { transform: 'rotate(-360deg)'},
        },
        "shake": {
          "10%":{
            transform: "translate3d(-1px, 0, 0)",
          },
          
          "20%":{
            transform: "translate3d(2px, 0, 0)",
          },
        
          "30%":{
            transform: "translate3d(-4px, 0, 0)",
          },
        
          "40%":{
            transform: "translate3d(4px, 0, 0)",
          },

          "50%":{
            transform: "translate3d(-4px, 0, 0)",
          },
        
          "60%":{
            transform: "translate3d(4px, 0, 0)",
          },

          "70%":{
            transform: "translate3d(4px, 0, 0)",
          },

          "80%":{
            transform: "translate3d(2px, 0, 0)",
          },

          "90%":{
            transform: "translate3d(-1px, 0, 0)",
          },
          
        },
        "scale": {
          from: { transform: 'scale(0)'},
          to: { transform: 'scale(1)'},
        }
      },
      animation: {
        'slow-appear': 'slow-appear 0.25s linear',
        "slow-appear-no-opacity": 'slow-appear-no-opacity 0.25s linear',
        'one-roll': 'roll .25s linear',
        'fast-shake': 'shake .25s linear',
        'scale': 'scale .12s linear',
      },
    },
    fontFamily: {
      rubik: ['Rubik', 'sans-serif'],
    }
  },
  plugins:[
    require('@tailwindcss/container-queries'),
  ],
};
