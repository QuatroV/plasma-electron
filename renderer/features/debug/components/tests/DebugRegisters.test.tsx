import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import DebugRegisters from "../DebugRegisters";

describe("DebugRegisters", () => {
  // Hack, see link: https://github.com/facebook/react/issues/11565#issuecomment-380143358
  beforeAll(() => {
    //@ts-ignore
    ReactDOM.createPortal = jest.fn((element, node) => {
      return element;
    });
  });

  afterEach(() => {
    //@ts-ignore
    ReactDOM.createPortal.mockClear();
  });
  it("renders correctly", () => {
    const tree = renderer.create(<DebugRegisters />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
