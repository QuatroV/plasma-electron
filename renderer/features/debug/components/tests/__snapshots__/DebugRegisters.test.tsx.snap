// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DebugRegisters renders correctly 1`] = `
<div>
  <div
    className="px-2 py-1 text-sm font-bold"
  >
    Registers
  </div>
  <div
    className="px-2"
  >
    <table
      className="w-full table-fixed  text-xs"
    >
      <thead
        className="  text-xs uppercase text-gray-700 shadow dark:bg-gray-700 dark:text-gray-400"
      >
        <tr>
          <th
            className=" rounded-tl bg-gray-50 py-1 pl-4 text-start"
          >
            Register
          </th>
          <th
            className="  flex-row items-center justify-between rounded-tr bg-gray-50 py-1 pl-4 text-start"
          >
            <div>
              Hex
            </div>
          </th>
          <th
            className="  flex-row items-center justify-between rounded-tr bg-gray-50 py-1 pl-4 text-start"
          >
            <div>
              Info
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        className="border border-gray-300"
      >
        <tr
          className="bg-gray-200 cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300"
        >
          <td
            className="pl-4"
          >
            EAX
          </td>
          <td
            className="pl-4"
          />
          <td
            className="pl-4"
          />
        </tr>
        <tr
          className="bg-gray-100 cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300"
        >
          <td
            className="pl-4"
          >
            EBX
          </td>
          <td
            className="pl-4"
          />
          <td
            className="pl-4"
          />
        </tr>
        <tr
          className="bg-gray-200 cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300"
        >
          <td
            className="pl-4"
          >
            ECX
          </td>
          <td
            className="pl-4"
          />
          <td
            className="pl-4"
          />
        </tr>
        <tr
          className="bg-gray-100 cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300"
        >
          <td
            className="pl-4"
          >
            EDX
          </td>
          <td
            className="pl-4"
          />
          <td
            className="pl-4"
          />
        </tr>
        <tr
          className="bg-gray-200 cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300"
        >
          <td
            className="pl-4"
          >
            ESI
          </td>
          <td
            className="pl-4"
          />
          <td
            className="pl-4"
          />
        </tr>
        <tr
          className="bg-gray-100 cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300"
        >
          <td
            className="pl-4"
          >
            EDI
          </td>
          <td
            className="pl-4"
          />
          <td
            className="pl-4"
          />
        </tr>
        <tr
          className="bg-gray-200 cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300"
        >
          <td
            className="pl-4"
          >
            ESP
          </td>
          <td
            className="pl-4"
          />
          <td
            className="pl-4"
          />
        </tr>
        <tr
          className="bg-gray-100 cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300"
        >
          <td
            className="pl-4"
          >
            EBP
          </td>
          <td
            className="pl-4"
          />
          <td
            className="pl-4"
          />
        </tr>
        <tr
          className="bg-gray-200 cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300"
        >
          <td
            className="pl-4"
          >
            CS
          </td>
          <td
            className="pl-4"
          />
          <td
            className="pl-4"
          />
        </tr>
        <tr
          className="bg-gray-100 cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300"
        >
          <td
            className="pl-4"
          >
            SS
          </td>
          <td
            className="pl-4"
          />
          <td
            className="pl-4"
          />
        </tr>
        <tr
          className="bg-gray-200 cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300"
        >
          <td
            className="pl-4"
          >
            DS
          </td>
          <td
            className="pl-4"
          />
          <td
            className="pl-4"
          />
        </tr>
        <tr
          className="bg-gray-100 cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300"
        >
          <td
            className="pl-4"
          >
            ES
          </td>
          <td
            className="pl-4"
          />
          <td
            className="pl-4"
          />
        </tr>
        <tr
          className="bg-gray-200 cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300"
        >
          <td
            className="pl-4"
          >
            FS
          </td>
          <td
            className="pl-4"
          />
          <td
            className="pl-4"
          />
        </tr>
        <tr
          className="bg-gray-100 cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300"
        >
          <td
            className="pl-4"
          >
            GS
          </td>
          <td
            className="pl-4"
          />
          <td
            className="pl-4"
          />
        </tr>
        <tr
          className="bg-gray-200 cursor-pointer border-b border-solid border-gray-300 transition-all hover:bg-gray-300"
        >
          <td
            className="pl-4"
          >
            EIP
          </td>
          <td
            className="pl-4"
          />
          <td
            className="pl-4"
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
